### ISSUE TITLE
"No such file or directory - gcloud" warning on Ruby 2.6.1

### ISSUE THREAD

**Issue by morgoth** [googleapis/google-auth-library-ruby#192]
I updated Rails app to Ruby 2.6.1 and when loading this lib code, I get warning in stdout output: `No such file or directory - gcloud`

This is because of this: https://github.com/googleapis/google-auth-library-ruby/blob/master/lib/googleauth/credentials_loader.rb#L176-L177 when process tries to run `gcloud` command on my system.

Looks like Ruby < 2.6.0 just silently swallowed this output message

I'm using googleauth 0.8.0

**Comment by TheRoyalTnetennba** [comment]
What OS are you using?

**Comment by morgoth** [comment]
It's Ubuntu 18.10
`Linux tp 4.18.0-13-generic #14-Ubuntu SMP Wed Dec 5 09:04:24 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux`

**Comment by TheRoyalTnetennba** [comment]
The only way I was able to reproduce this is by passing the --debug argument. Can you post the output of  `echo $RUBYOPT`?

**Comment by morgoth** [comment]
Inside Rails app it's "-rbundler/setup".
I can see a difference in irb and rails console:
```
irb
irb(main):001:0> require 'googleauth'
=> true
irb(main):002:0> Google::Auth::CredentialsLoader.load_gcloud_project_id
=> nil
irb(main):003:0> exit
bin/rails c
Running via Spring preloader in process 17860
Loading development environment (Rails 5.2.2)
[1] pry(main)> Google::Auth::CredentialsLoader.load_gcloud_project_id
rails_console: No such file or directory - gcloud
=> nil
```
Not sure what exactly is happening, but there was no such output when using Ruby 2.6.0

**Comment by naerymdan** [comment]
I also have this exact issue, but it seems to be less about the ruby version rather a regression in googleauth 0.7+:

On `googleauth 0.6.7` I can run google libraries like `google-cloud-storage` and `google-cloud-pubsub` on a Debian or Alpine container on GKE and get implied credentials.

Starting from 0.7 and still in 0.8, I cannot get credentials in my containers, and it logs me `bin/rails: No such file or directory - gcloud` or `bin/rake: No such file or directory - gcloud` by hitting the exact `load_gcloud_project_id` function that tries to call the gcloud executable, which obviously does not exist in a docker container.

**Comment by morgoth** [comment]
This can be reproduced on newly generated Rails app (5.2.2) on Ruby 2.6.1

**Comment by BintLopez** [comment]
I am also experiencing this with a Rails app 5.1.5 with Ruby 2.5.3. Note -- I'm only getting this when running the app in Heroku. Locally, it doesn't throw this error.

The way I'm calling auth is as follows where SCOPES is an array of strings.
```ruby
::Google::Auth::ServiceAccountCredentials.make_creds(scope: SCOPES)
```

The error it throws is
```
/app/bin/rails: No such file or directory - gcloud
Unable to determine project id.
Signet::AuthorizationError (Authorization failed.  Server message:
{
  error": "unsupported_grant_type",
  "error_description": "Invalid grant_type: "
})
```

**Comment by TheRoyalTnetennba** [comment]
How are you building credentials?

**Comment by TheRoyalTnetennba** [comment]
Super interesting. Seems like rails at least used to [overwrite backticks](https://github.com/rails/rails/blob/v5.1.4/activesupport/lib/active_support/core_ext/kernel/agnostics.rb), which is how the gcloud command is called. Switching from backticks to IO.popen seems to fix it. https://github.com/googleapis/google-auth-library-ruby/pull/194

**Comment by TheRoyalTnetennba** [comment]
Super interesting. Seems like rails at least used to [overwrite backticks](https://github.com/rails/rails/blob/v5.1.4/activesupport/lib/active_support/core_ext/kernel/agnostics.rb), which is how the gcloud command is called. Switching from backticks to IO.popen seems to fix it. https://github.com/googleapis/google-auth-library-ruby/pull/194

**Comment by morgoth** [comment]
Thanks for this finding. This overwrite was removed from Rails https://github.com/rails/rails/pull/31253 but it will only be released in Rails 6.0
